aidial_adapter_bedrock/llm/truncate_prompt.py

```diff
@@ -66,15 +66,41 @@ def partition_indexer(chunks: List[int]) -> Callable[[int], List[int]]:
     return mapping.__getitem__
 
 
-T = TypeVar("T")
-Messages = List[T]
+_T = TypeVar("_T")
 
 
-def truncate_prompt(
-    messages: Messages,
-    tokenize_messages: Callable[[Messages], int],
-    keep_message: Callable[[Messages, int], bool],
-    partition_messages: Callable[[Messages], List[int]],
+async def truncate_prompt(
+    messages: List[_T],
+    tokenize_messages: Callable[[List[_T]], Awaitable[int]],
+    keep_message: Callable[[List[_T], int], bool],
+    partition_messages: Callable[[List[_T]], List[int]],
```

Comment:
I don't follow the logic. Why should it be `partitioner`?

Here I called all callback as verbs by the action they do:
* `tokenize_messages` - it tokenized messages,
* `keep_message` - it preserves a message,
* `partition_messages` - it creates a [partition](https://en.wikipedia.org/wiki/Partition_of_a_set) of messages

Reply:
Because chain of calls here is such:

```python

class PseudoChatModel(...):
    partitioner: Callable[[List[BaseMessage]], List[int]]
    
     @override
    async def truncate_and_linearize_messages(
        self, messages: List[BaseMessage], max_prompt_tokens: Optional[int]
    ) -> TextCompletionPrompt:
        discarded_messages, messages = await truncate_prompt(
            ...
           partition_messages=self.partitioner,  <-------
       )
```

Meaning of this function has not changed - it's a function, that takes list of elements, and returns list of partition sizes

Why we need to change this naming then?

Let's choose `partition_messages` or `partitioner`, and use it consistently